pkg/cmd/logs.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/daytonaio/daytona/cmd/daytona/config"
+	"github.com/daytonaio/daytona/internal/util"
+	apiclient_util "github.com/daytonaio/daytona/internal/util/apiclient"
+	"github.com/daytonaio/daytona/pkg/apiclient"
+	"github.com/daytonaio/daytona/pkg/views"
+	"github.com/daytonaio/daytona/pkg/views/workspace/selection"
+	"github.com/spf13/cobra"
+)
+
+var followFlag bool
+var workspaceFlag bool
+
+var logsCmd = &cobra.Command{
+	Use:     "logs [WORKSPACE] [PROJECT_NAME]",
+	Short:   "View logs for a workspace/project",
+	Args:    cobra.RangeArgs(0, 2),
+	GroupID: util.WORKSPACE_GROUP,
+	Aliases: []string{"lg", "log"},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx := context.Background()
+		c, err := config.GetConfig()
+		if err != nil {
+			return err
+		}
+
+		activeProfile, err := c.GetActiveProfile()
+		if err != nil {
+			return err
+		}
+
+		var workspace *apiclient.WorkspaceDTO
+		apiClient, err := apiclient_util.GetApiClient(&activeProfile)
+		if err != nil {
+			return err
+		}
+
+		if len(args) == 0 {
+			workspaceList, res, err := apiClient.WorkspaceAPI.ListWorkspaces(ctx).Execute()
+			if err != nil {
+				return apiclient_util.HandleErrorResponse(res, err)
+			}
+			if len(workspaceList) == 0 {
+				views.RenderInfoMessage("The workspace list is empty. Start off by running 'daytona create'.")
+				return nil
+			}
+			workspace = selection.GetWorkspaceFromPrompt(workspaceList, "Get Logs For")
+		} else {
+			workspace, err = apiclient_util.GetWorkspace(args[0], false)
+			if err != nil {
+				return err
+			}
+		}
+
+		var (
+			projectName       string
```

Comment:
No need for `projectName`.

`projectNames` is enough.

This will also create the desired behavior when doing `daytona logs WS PROJ --workspace`.

![Screenshot 2024-10-02 at 10 29 32](https://github.com/user-attachments/assets/b4d2535d-e7f3-4fce-86fb-2a298b124a69)

Instead of only showing workspace logs in this case, it would show the project logs + workspace logs

Reply:
What should `dtn logs WS -w` return?

Reply:
Only the workspace logs.

`dtn logs WS -w` => only workspace logs
`dtn logs WS` => workspace + all project logs
`dtn logs WS PROJ` => only PROJ logs
`dtn logs WS PROJ -w` => workspace + PROJ logs

Reply:
Thanks for the clarity!

Reply:
Do you know why this is happening?
![Screenshot from 2024-10-02 16-13-18](https://github.com/user-attachments/assets/dea4927d-bad9-48e0-9cbb-abcc4b06a141)
![Screenshot from 2024-10-02 16-12-38](https://github.com/user-attachments/assets/d0ded08e-42c3-4dd5-9902-3b2cc00ebd9b)

Reply:
Do you know why this is happening? I'm trying to achieve `dtn logs WS -w`
![Screenshot from 2024-10-02 16-13-18](https://github.com/user-attachments/assets/68b8bae6-e45f-438d-8115-83893db2c497)
![Screenshot from 2024-10-02 16-12-38](https://github.com/user-attachments/assets/300b1568-9b78-43dc-a74c-85bddc2bd2da)

Reply:
Not sure, but don't think it's related to you. I'll check it out after you finish.

Reply:
Cool!

Reply:
```
daytona ➜ /workspaces/daytona (feat/logs) $ dtn logs test-repo -w
 WORKSPACE | ✓ Creating workspace test-repo (6babd87a9585)
 WORKSPACE | ✓ Workspace creation complete. Pending start...
 WORKSPACE | ✓ Starting workspace
 WORKSPACE | ✓ Workspace test-repo started
 WORKSPACE | ✓ %                                                                                                                                                                          
daytona ➜ /workspaces/daytona (feat/logs) $ dtn logs test-repo   
 WORKSPACE   | ✓ Creating workspace test-repo (6babd87a9585)
 WORKSPACE   | ✓ Workspace creation complete. Pending start...
 WORKSPACE   | ✓ Starting workspace
 WORKSPACE   | ✓ Workspace test-repo started
 test-repo   | Creating project test-repo
 test-repo-2 | Creating project test-repo-2
 test-repo   | Pulling image...
 test-repo-2 | Pulling image...
 test-repo   | Pulling from daytonaio/workspace-project
 test-repo   | Digest: sha256:adf39125cf348f8801b8e1d9be37d0693eade0602344b42908433b5c9be1c7ad
 test-repo-2 | Pulling from daytonaio/workspace-project
 test-repo   | Status: Image is up to date for daytonaio/workspace-project:latest
 test-repo   | Image pulled successfully
 test-repo   | UIDs and GIDs are the same (1000:1000).
 test-repo-2 | Digest: sha256:adf39125cf348f8801b8e1d9be37d0693eade0602344b42908433b5c9be1c7ad
 test-repo   | Cloning into '/workdir/6babd87a9585-test-repo'...
 test-repo-2 | Status: Image is up to date for daytonaio/workspace-project:latest
 test-repo   | warning: redirecting to https://github.com/unsuman/test-repo.git/
 test-repo   | UIDs and GIDs are the same (1000:1000).
 test-repo   | Project test-repo created
 test-repo   | Starting project test-repo
 test-repo-2 | Image pulled successfully
 test-repo-2 | UIDs and GIDs are the same (1000:1000).
 test-repo   | Found devcontainer.metadata label
 test-repo   | Executed entrypoint for feature ghcr.io/devcontainers/features/docker-in-docker:2.11.0
 test-repo-2 | Cloning into '/workdir/6babd87a9585-test-repo-2'...
 test-repo-2 | warning: redirecting to https://github.com/unsuman/test-repo.git/
 test-repo   | Executed entrypoint for feature ghcr.io/devcontainers/features/nix:1.2.0
 test-repo   | Downloading Daytona from http://host.docker.internal:3986/binary/v0.0.0-dev/daytona-linux-amd64
 test-repo-2 | UIDs and GIDs are the same (1000:1000).
 test-repo-2 | Project test-repo-2 created
 test-repo   | Daytona downloaded successfully
 test-repo   | Installing server to /usr/local/bin
 test-repo   | Starting Daytona Agent
 test-repo-2 | Starting project test-repo-2
 test-repo   | Repository already exists. Skipping clone...
 test-repo-2 | Found devcontainer.metadata label
 test-repo   | failed to get git user data: Request to https://api.github.com/user failed with Requires authentication
 test-repo-2 | Executed entrypoint for feature ghcr.io/devcontainers/features/docker-in-docker:2.11.0
 test-repo   | Starting ssh server on port 2222...
 test-repo-2 | Executed entrypoint for feature ghcr.io/devcontainers/features/nix:1.2.0
 test-repo-2 | Downloading Daytona from http://host.docker.internal:3986/binary/v0.0.0-dev/daytona-linux-amd64
 test-repo   | Project test-repo started
 test-repo-2 | Daytona downloaded successfully
 test-repo-2 | Installing server to /usr/local/bin
 test-repo-2 | Starting Daytona Agent
 test-repo-2 | Repository already exists. Skipping clone...
 test-repo-2 | failed to get git user data: Request to https://api.github.com/user failed with Requires authentication
 test-repo-2 | Starting ssh server on port 2222...
 test-repo-2 | Project test-repo-2 started
daytona ➜ /workspaces/daytona (feat/logs) $ dtn logs test-repo -w
 WORKSPACE | ✓ Creating workspace test-repo (6babd87a9585)
 WORKSPACE | ✓ Workspace creation complete. Pending start...
 WORKSPACE | ✓ Starting workspace
 WORKSPACE | ✓ Workspace test-repo started
 WORKSPACE | ✓ %                                                                                                                                                                          
daytona ➜ /workspaces/daytona (feat/logs) $ 
daytona ➜ /workspaces/daytona (feat/logs) $ dtn logs test-repo   
 WORKSPACE   | ✓ Creating workspace test-repo (6babd87a9585)
 WORKSPACE   | ✓ Workspace creation complete. Pending start...
 WORKSPACE   | ✓ Starting workspace
 WORKSPACE   | ✓ Workspace test-repo started
 test-repo   | Creating project test-repo
 test-repo-2 | Creating project test-repo-2
 test-repo   | Pulling image...
 test-repo-2 | Pulling image...
 test-repo   | Pulling from daytonaio/workspace-project
 test-repo-2 | Pulling from daytonaio/workspace-project
 test-repo   | Digest: sha256:adf39125cf348f8801b8e1d9be37d0693eade0602344b42908433b5c9be1c7ad
 test-repo-2 | Digest: sha256:adf39125cf348f8801b8e1d9be37d0693eade0602344b42908433b5c9be1c7ad
 test-repo   | Status: Image is up to date for daytonaio/workspace-project:latest
 test-repo   | Image pulled successfully
 test-repo-2 | Status: Image is up to date for daytonaio/workspace-project:latest
 test-repo-2 | Image pulled successfully
 test-repo   | UIDs and GIDs are the same (1000:1000).
 test-repo-2 | UIDs and GIDs are the same (1000:1000).
 test-repo   | Cloning into '/workdir/6babd87a9585-test-repo'...
 test-repo-2 | Cloning into '/workdir/6babd87a9585-test-repo-2'...
 test-repo   | warning: redirecting to https://github.com/unsuman/test-repo.git/
 test-repo-2 | warning: redirecting to https://github.com/unsuman/test-repo.git/
 test-repo   | UIDs and GIDs are the same (1000:1000).
 test-repo-2 | UIDs and GIDs are the same (1000:1000).
 test-repo   | Project test-repo created
 test-repo   | Starting project test-repo
 test-repo   | Found devcontainer.metadata label
 test-repo-2 | Project test-repo-2 created
 test-repo-2 | Starting project test-repo-2
 test-repo   | Executed entrypoint for feature ghcr.io/devcontainers/features/docker-in-docker:2.11.0
 test-repo   | Executed entrypoint for feature ghcr.io/devcontainers/features/nix:1.2.0
 test-repo   | Downloading Daytona from http://host.docker.internal:3986/binary/v0.0.0-dev/daytona-linux-amd64
 test-repo-2 | Found devcontainer.metadata label
 test-repo   | Daytona downloaded successfully
 test-repo-2 | Executed entrypoint for feature ghcr.io/devcontainers/features/docker-in-docker:2.11.0
 test-repo-2 | Executed entrypoint for feature ghcr.io/devcontainers/features/nix:1.2.0
 test-repo   | Installing server to /usr/local/bin
 test-repo   | Starting Daytona Agent
 test-repo-2 | Downloading Daytona from http://host.docker.internal:3986/binary/v0.0.0-dev/daytona-linux-amd64
 test-repo   | Repository already exists. Skipping clone...
 test-repo   | failed to get git user data: Request to https://api.github.com/user failed with Requires authentication
 test-repo-2 | Daytona downloaded successfully
 test-repo   | Starting ssh server on port 2222...
 test-repo-2 | Installing server to /usr/local/bin
 test-repo   | Project test-repo started
 test-repo-2 | Starting Daytona Agent
 test-repo-2 | Repository already exists. Skipping clone...
 test-repo-2 | failed to get git user data: Request to https://api.github.com/user failed with Requires authentication
 test-repo-2 | Starting ssh server on port 2222...
 test-repo-2 | Project test-repo-2 started
daytona ➜ /workspaces/daytona (feat/logs) $ dtn logs test-repo test-repo-2
 test-repo-2 | Creating project test-repo-2
 test-repo-2 | Pulling image...
 test-repo-2 | Pulling from daytonaio/workspace-project
 test-repo-2 | Digest: sha256:adf39125cf348f8801b8e1d9be37d0693eade0602344b42908433b5c9be1c7ad
 test-repo-2 | Status: Image is up to date for daytonaio/workspace-project:latest
 test-repo-2 | Image pulled successfully
 test-repo-2 | UIDs and GIDs are the same (1000:1000).
 test-repo-2 | Cloning into '/workdir/6babd87a9585-test-repo-2'...
 test-repo-2 | warning: redirecting to https://github.com/unsuman/test-repo.git/
 test-repo-2 | UIDs and GIDs are the same (1000:1000).
 test-repo-2 | Project test-repo-2 created
 test-repo-2 | Starting project test-repo-2
 test-repo-2 | Found devcontainer.metadata label
 test-repo-2 | Executed entrypoint for feature ghcr.io/devcontainers/features/docker-in-docker:2.11.0
 test-repo-2 | Executed entrypoint for feature ghcr.io/devcontainers/features/nix:1.2.0
 test-repo-2 | Downloading Daytona from http://host.docker.internal:3986/binary/v0.0.0-dev/daytona-linux-amd64
 test-repo-2 | Daytona downloaded successfully
 test-repo-2 | Installing server to /usr/local/bin
 test-repo-2 | Starting Daytona Agent
 test-repo-2 | Repository already exists. Skipping clone...
 test-repo-2 | failed to get git user data: Request to https://api.github.com/user failed with Requires authentication
 test-repo-2 | Starting ssh server on port 2222...
 test-repo-2 | Project test-repo-2 started
daytona ➜ /workspaces/daytona (feat/logs) $ dtn logs test-repo test-repo-2
 test-repo-2 | Creating project test-repo-2
 test-repo-2 | Pulling image...
 test-repo-2 | Pulling from daytonaio/workspace-project
 test-repo-2 | Digest: sha256:adf39125cf348f8801b8e1d9be37d0693eade0602344b42908433b5c9be1c7ad
 test-repo-2 | Status: Image is up to date for daytonaio/workspace-project:latest
 test-repo-2 | Image pulled successfully
 test-repo-2 | UIDs and GIDs are the same (1000:1000).
 test-repo-2 | Cloning into '/workdir/6babd87a9585-test-repo-2'...
 test-repo-2 | warning: redirecting to https://github.com/unsuman/test-repo.git/
 test-repo-2 | UIDs and GIDs are the same (1000:1000).
 test-repo-2 | Project test-repo-2 created
 test-repo-2 | Starting project test-repo-2
 test-repo-2 | Found devcontainer.metadata label
 test-repo-2 | Executed entrypoint for feature ghcr.io/devcontainers/features/docker-in-docker:2.11.0
 test-repo-2 | Executed entrypoint for feature ghcr.io/devcontainers/features/nix:1.2.0
 test-repo-2 | Downloading Daytona from http://host.docker.internal:3986/binary/v0.0.0-dev/daytona-linux-amd64
 test-repo-2 | Daytona downloaded successfully
 test-repo-2 | Installing server to /usr/local/bin
 test-repo-2 | Starting Daytona Agent
 test-repo-2 | Repository already exists. Skipping clone...
 test-repo-2 | failed to get git user data: Request to https://api.github.com/user failed with Requires authentication
 test-repo-2 | Starting ssh server on port 2222...
 test-repo-2 | Project test-repo-2 started
daytona ➜ /workspaces/daytona (feat/logs) $ dtn logs test-repo test-repo-2 -w
 WORKSPACE   | ✓ Creating workspace test-repo (6babd87a9585)
 WORKSPACE   | ✓ Workspace creation complete. Pending start...
 WORKSPACE   | ✓ Starting workspace
 WORKSPACE   | ✓ Workspace test-repo started
 test-repo-2 | Creating project test-repo-2
 test-repo-2 | Pulling image...
 test-repo-2 | Pulling from daytonaio/workspace-project
 test-repo-2 | Digest: sha256:adf39125cf348f8801b8e1d9be37d0693eade0602344b42908433b5c9be1c7ad
 test-repo-2 | Status: Image is up to date for daytonaio/workspace-project:latest
 test-repo-2 | Image pulled successfully
 test-repo-2 | UIDs and GIDs are the same (1000:1000).
 test-repo-2 | Cloning into '/workdir/6babd87a9585-test-repo-2'...
 test-repo-2 | warning: redirecting to https://github.com/unsuman/test-repo.git/
 test-repo-2 | UIDs and GIDs are the same (1000:1000).
 test-repo-2 | Project test-repo-2 created
 test-repo-2 | Starting project test-repo-2
 test-repo-2 | Found devcontainer.metadata label
 test-repo-2 | Executed entrypoint for feature ghcr.io/devcontainers/features/docker-in-docker:2.11.0
 test-repo-2 | Executed entrypoint for feature ghcr.io/devcontainers/features/nix:1.2.0
 test-repo-2 | Downloading Daytona from http://host.docker.internal:3986/binary/v0.0.0-dev/daytona-linux-amd64
 test-repo-2 | Daytona downloaded successfully
 test-repo-2 | Installing server to /usr/local/bin
 test-repo-2 | Starting Daytona Agent
 test-repo-2 | Repository already exists. Skipping clone...
 test-repo-2 | failed to get git user data: Request to https://api.github.com/user failed with Requires authentication
 test-repo-2 | Starting ssh server on port 2222...
 test-repo-2 | Project test-repo-2 started
daytona ➜ /workspaces/daytona (feat/logs) $ dtn logs test-repo sdadasd       
FATA[0000] project not found in workspace               
exit status 1
```